cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,1070 @@
+  type: string
+  sample: "nat-0d1e3a878585988f8"
+subnet_id:
+  description: id of the Subnet
+  returned: In all cases.
+  type: string
+  sample: "subnet-12345"
+state:
+  description: The current state of the Nat Gateway.
+  returned: In all cases.
+  type: string
+  sample: "available"
+vpc_id:
+  description: id of the VPC.
+  returned: In all cases.
+  type: string
+  sample: "vpc-12345"
+nat_gateway_addresses:
+  description: List of dictionairies containing the public_ip, network_interface_id, private_ip, and allocation_id.
+  returned: In all cases.
+  type: string
+  sample: [
+      {
+          'public_ip': '52.52.52.52',
+          'network_interface_id': 'eni-12345',
+          'private_ip': '10.0.0.100',
+          'allocation_id': 'eipalloc-12345'
+      }
+  ]
+'''
+
+try:
+    import botocore
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+import datetime
+import random
+import time
+
+from dateutil.tz import tzutc
+
+DRY_RUN_GATEWAYS = [
+    {
+        "nat_gateway_id": "nat-123456789",
+        "subnet_id": "subnet-123456789",
+        "nat_gateway_addresses": [
+            {
+                "public_ip": "55.55.55.55",
+                "network_interface_id": "eni-1234567",
+                "private_ip": "10.0.0.102",
+                "allocation_id": "eipalloc-1234567"
+            }
+        ],
+        "state": "available",
+        "create_time": "2016-03-05T05:19:20.282000+00:00",
+        "vpc_id": "vpc-12345678"
+    }
+]
+DRY_RUN_GATEWAY_UNCONVERTED = [
+    {
+        'VpcId': 'vpc-12345678',
+        'State': 'available',
+        'NatGatewayId': 'nat-123456789',
+        'SubnetId': 'subnet-123456789',
+        'NatGatewayAddresses': [
+            {
+                'PublicIp': '55.55.55.55',
+                'NetworkInterfaceId': 'eni-1234567',
+                'AllocationId': 'eipalloc-1234567',
+                'PrivateIp': '10.0.0.102'
+            }
+        ],
+        'CreateTime': datetime.datetime(2016, 3, 5, 5, 19, 20, 282000, tzinfo=tzutc())
+    }
+]
+
+DRY_RUN_ALLOCATION_UNCONVERTED = {
+    'Addresses': [
+        {
+            'PublicIp': '55.55.55.55',
+            'Domain': 'vpc',
+            'AllocationId': 'eipalloc-1234567'
+        }
+    ]
+}
+
+DRY_RUN_MSGS = 'DryRun Mode:'
+
+def convert_to_lower(data):
+    """Convert all uppercase keys in dict with lowercase_
+
+    Args:
+        data (dict): Dictionary with keys that have upper cases in them
+            Example.. FooBar == foo_bar
+            if a val is of type datetime.datetime, it will be converted to
+            the ISO 8601
+
+    Basic Usage:
+        >>> test = {'FooBar': []}
+        >>> test = convert_to_lower(test)
+        {
+            'foo_bar': []
+        }
+
+    Returns:
+        Dictionary
+    """
+    results = dict()
+    if isinstance(data, dict):
+        for key, val in data.items():
+            key = re.sub(r'(([A-Z]{1,3}){1})', r'_\1', key).lower()
```

Comment:
This works as 're' is import from basic, but it's preferred to add a 'import re' for clarity